src/test/java/BMSStreamSender/BatteryManagementSystemTest.java

```diff
@@ -0,0 +1,47 @@
+package BMSStreamSender;
+import static org.junit.Assert.*;
+
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.junit.Test;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BatteryManagementSystemTest {
+
+	@Test
+	 public void generateParamValues_whenValueIsEmpty_thenReturnFalse() {
+			
+		 assertFalse(ValidateUserInput.isNumeric(""));
+	 }
+	
+	@Test
+	 public void generateParamValues_whenValueInValid_thenReturnFalse() {
+			
+		 assertFalse(ValidateUserInput.isNumeric("12B"));
+	 }
+	
+	@Test
+	public void generateParamValues_whenValueIsLesserThan10_thenReturnFalse() {
+		assertFalse(ValidateUserInput.isGreaterthanMin("5"));
+	}
+	
+	@Test
+	public void generateParamValues_whenValueIsGreaterOrEquals10_thenReturnTrue() {
+		assertTrue(ValidateUserInput.isGreaterthanMin("10"));
+	}
+	
+	@Test
+	public void generateParamValues_whenGeneratedParamValCount_thenReturnTrue() {
+		IBMSService bms_service=new BMSServiceImpl();
+		int expected_reading_count=12;
+		assertEquals(bms_service.genearateBatteryReadings(12).size(),expected_reading_count);
+	}
+	
+	@Test
+	public void checkIfBMSValuesPrinting() {
+	IBMSService bmsStreamerImplMock = Mockito.mock(BMSServiceImpl.class);
+	((BMSServiceImpl) bmsStreamerImplMock).printBatteryReadingsToConsole();
+	((BMSServiceImpl) Mockito.verify(bmsStreamerImplMock, Mockito.times(1))).printBatteryReadingsToConsole();
```

Comment:
This will just verify the function call. Not the format of output. In future, if someone changes the format of the output by mistake, the error will only be caught when it's integrated with the receiver. The purpose of unit tests is to catch defects early. Only a few such defects are caught early by these unit tests.